src/psen_scan_driver.cpp

```diff
@@ -52,17 +67,19 @@ int main(int argc, char** argv)
 
   try
   {
-    psen_scan_v2::RosParameterHandler param_handler(pnh);
-
-    ScannerConfiguration scanner_configuration(param_handler.getHostIP(),
-                                               param_handler.getHostUDPPortData(),
-                                               param_handler.getHostUDPPortControl(),
-                                               param_handler.getSensorIP(),
-                                               DEFAULT_X_AXIS_ROTATION - param_handler.getAngleEnd(),
-                                               DEFAULT_X_AXIS_ROTATION - param_handler.getAngleStart());
-
-    ROSScannerNode ros_scanner_node(
-        pnh, DEFAULT_PUBLISH_TOPIC, param_handler.getFrameID(), DEFAULT_X_AXIS_ROTATION, scanner_configuration);
+    ScannerConfiguration scanner_configuration(
+        getRequiredParamFromServer<std::string>(pnh, PARAM_HOST_IP),
+        getRequiredParamFromServer<int>(pnh, PARAM_HOST_DATA_PORT),
+        getRequiredParamFromServer<int>(pnh, PARAM_HOST_CONTROL_PORT),
+        getRequiredParamFromServer<std::string>(pnh, PARAM_SCANNER_IP),
+        DEFAULT_X_AXIS_ROTATION - getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_END, DEFAULT_ANGLE_END),
+        DEFAULT_X_AXIS_ROTATION - getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_START, DEFAULT_ANGLE_START));
+
+    ROSScannerNode ros_scanner_node(pnh,
+                                    DEFAULT_PUBLISH_TOPIC,
+                                    getOptionalParamFromServer<std::string>(pnh, PARAM_FRAME_ID, DEFAULT_FRAME_ID),
+                                    DEFAULT_X_AXIS_ROTATION,
+                                    scanner_configuration);
```

Comment:
```suggestion
    const auto host_ip{ getRequiredParamFromServer<std::string>(pnh, PARAM_HOST_IP) };
    const auto host_data_port{ getRequiredParamFromServer<int>(pnh, PARAM_HOST_DATA_PORT) };
    const auto host_control_port{ getRequiredParamFromServer<int>(pnh, PARAM_HOST_CONTROL_PORT) };
    const auto scanner_ip{ getRequiredParamFromServer<std::string>(pnh, PARAM_SCANNER_IP) };
    const auto angle_start{ DEFAULT_X_AXIS_ROTATION -
                            getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_END, DEFAULT_ANGLE_END) };
    const auto angle_end{ DEFAULT_X_AXIS_ROTATION -
                          getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_START, DEFAULT_ANGLE_START) };

    ScannerConfiguration scanner_configuration(
        host_ip, host_data_port, host_control_port, scanner_ip, angle_start, angle_end);

    const auto frame_id{ getOptionalParamFromServer<std::string>(pnh, PARAM_FRAME_ID, DEFAULT_FRAME_ID) };

    ROSScannerNode ros_scanner_node(
        pnh, DEFAULT_PUBLISH_TOPIC, frame_id, DEFAULT_X_AXIS_ROTATION, scanner_configuration);
```

Reply:
I was wondering if the readability is better if the parameters or initialized on separate lines. What do you think. At least concerning the start-/end-angle I would prefer it like that.

Reply:
I opted for a compromise: I initialized the parameters start- and end-angle on a separate line.